src/Hosting/Hosting/src/StaticWebAssets/StaticWebAssetsFileProvider.cs

```diff
@@ -40,20 +44,29 @@ public StaticWebAssetsFileProvider(string pathPrefix, string contentRoot)
         /// <inheritdoc />
         public IDirectoryContents GetDirectoryContents(string subpath)
         {
-            if (!StartsWithBasePath(subpath, out var physicalPath))
+            var modifiedSub = subpath.StartsWith("/") ? subpath : "/" + subpath;
+
+            if (StartsWithBasePath(modifiedSub, out var physicalPath))
             {
-                return NotFoundDirectoryContents.Singleton;
+                return InnerProvider.GetDirectoryContents(physicalPath);
             }
             else
+            if (string.Equals(subpath, string.Empty) || BasePath.StartsWithSegments(modifiedSub))
             {
-                return InnerProvider.GetDirectoryContents(physicalPath);
+                return new FakeRoot(BasePath, modifiedSub);
+            }
```

Comment:
The `out remaining` part is nice (didn't notice that overload), but passing in the inner provider doesn't do anything for us. `IDirectoryContents` is an `IEnumerable<IFileInfo>`, and `IFileInfo doesn't have any way of indicating its children, so it can't model a linked list. What happens here is that MatcherContext [walks the IFileProvider results recursively](https://github.com/aspnet/Extensions/blob/master/src/FileSystemGlobbing/src/Internal/MatcherContext.cs#L121), so the InnerProvider will only get hit on a new call to GetDirectoryContents which completes the whole BasePath. That's also why I don't think it's useful to create a linked list inside `StaticWebAssetsDirectoryRoot` (formerly `FakeRoot`), because it's only going to use the first result regardless.

Reply:
That won't work.

You need to link the base path to the results of the file provider.

Reply:
Looking at another issue (https://github.com/aspnet/AspNetCore/issues/15278) I've realized that we don't support windows style paths in the static web assets file provider. I'm leaving the comment here so I can attach it to a new issue, and triage it separately.